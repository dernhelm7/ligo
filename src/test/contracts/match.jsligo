type storage = int;

type Add = {
  kind: 'Add',
  value: int
}
type Sub = {
  kind: 'Sub',
  value: int
}

type parameter =
| Add
| Sub;

type return = [list <operation>, storage];

let main = ([action, store]: [parameter, storage]) => {
  switch (action.kind) {
    case 'Add': {
      let store = store + action.value;
      return [([] as list <operation>), store];
    }
    case 'Sub': {
      let store = store - action.value;
      return [([] as list <operation>), store];
    }
  }
}
