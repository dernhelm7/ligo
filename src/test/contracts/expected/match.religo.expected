type storage = int;

type parameter = Add(int) | Sub(int);

type return = (list(operation), storage);

let main = ((action, store): (parameter, storage)) => {
  let store = 
    store + (switch(action){
       | Add(n) => n
       | Sub(n) => -n
       });
  (([] : list(operation)), store)
};
